x-pack/plugins/ml/server/models/data_frame_analytics/analytics_manager.ts

```diff
@@ -106,12 +116,8 @@ export class AnalyticsManager {
     );
   }
 
-  private findJob(id: string): estypes.MlDataframeAnalyticsSummary {
-    const job = this._jobs.find((js) => js.id === id);
-    if (job === undefined) {
-      throw Error(`No known job with id '${id}'`);
-    }
-    return job;
+  private findJob(id: string): estypes.MlDataframeAnalyticsSummary | undefined {
```

Comment:
For the edge case described in https://github.com/elastic/kibana/issues/164626, where the original job is deleted but then later a new job with the same name is created, is there any way for the job config to reference the ID of the model once it's been created?

Reply:
As of now, the job data doesn't have a reference to the model created from it. This might be something worth adding in the backend - we could check if it's available and confirm if it is. It would potentially also make this map code simpler.

Reply:
Let's just live with this limitation for now and continue with the existing logic for finding the model ID from the job ID even if in this edge case it will return the ID of the recreated job.